src/databricks/labs/ucx/cli.py

```diff
@@ -705,8 +705,16 @@ def migrate_acls(
         workspace_contexts = [ctx]
     else:
         workspace_contexts = _get_workspace_contexts(w, a, run_as_collection, **named_parameters)
-    target_catalog, hms_fed = named_parameters.get("target_catalog"), named_parameters.get("hms_fed", False)
+    target_catalog = named_parameters.get("target_catalog")
+    hms_fed = named_parameters.get("hms_fed", False)
+    dry_run = named_parameters.get("dry_run", False)
     for workspace_context in workspace_contexts:
+        if dry_run:
+            grants = workspace_context.acl_migrator.snapshot()
+            logger.info(
+                f"Dry run completed. Found {len(grants)} grants. The crawled grants can be found in the 'source_table_grants' table. "
+                "No changes were made."
+            )
```

Comment:
this still results in a call to `workspace_context.acl_migrator.migrate_acls`. and technically, it works only for the first workspace out of many. proper dry run has to be in a separate loop.

Reply:
Can you explain?

Reply:
Got it